eyeball-im-util/tests/it/sort_by.rs

```diff
@@ -0,0 +1,411 @@
+
+    // Pop back again.
+    assert_eq!(ob.pop_back(), Some('d'));
+    assert_next_eq!(sub, VectorDiff::Remove { index: 1 });
+
+    // Pop back again.
+    assert_eq!(ob.pop_back(), Some('a'));
+    assert_next_eq!(sub, VectorDiff::PopBack);
+
+    // Pop back again.
+    assert_eq!(ob.pop_back(), Some('b'));
+    assert_next_eq!(sub, VectorDiff::PopBack);
+
+    // Pop back again.
+    assert_eq!(ob.pop_back(), Some('e'));
+    assert_next_eq!(sub, VectorDiff::PopFront);
+
+    assert!(ob.is_empty());
+
+    drop(ob);
+    assert_closed!(sub);
+}
+
+#[test]
+fn remove() {
+    let mut ob = ObservableVector::<char>::new();
+    let (values, mut sub) = ob.subscribe().sort_by(&rev_cmp);
+
+    assert!(values.is_empty());
+    assert_pending!(sub);
+
+    // Append a bunch of items.
+    ob.append(vector!['e', 'b', 'a', 'd', 'c']);
+    assert_next_eq!(sub, VectorDiff::Append { values: vector!['e', 'd', 'c', 'b', 'a'] });
+
+    // Remove `a`.
+    ob.remove(2);
+    assert_next_eq!(sub, VectorDiff::PopBack);
+
+    // Remove `e`.
+    ob.remove(0);
+    assert_next_eq!(sub, VectorDiff::PopFront);
+
+    // Remove `c`.
+    ob.remove(2);
+    assert_next_eq!(sub, VectorDiff::Remove { index: 1 });
+
+    // Items in the vector have been removed and are not sorted.
+    assert_eq!(*ob, vector!['b', 'd']);
+
+    drop(ob);
+    assert_closed!(sub);
+}
+
+#[test]
+fn set() {
+    let mut ob = ObservableVector::<char>::new();
+    let (values, mut sub) = ob.subscribe().sort_by(&rev_cmp);
+
+    assert!(values.is_empty());
+    assert_pending!(sub);
+
+    // Append a bunch of items.
+    ob.append(vector!['d', 'e', 'b', 'g']);
+    assert_next_eq!(sub, VectorDiff::Append { values: vector!['g', 'e', 'd', 'b'] });
+
+    // Same value.
+    ob.set(0, 'd');
+    assert_next_eq!(sub, VectorDiff::Set { index: 2, value: 'd' });
+
+    // Another value, that is moved to the right.
+    ob.set(0, 'c');
+    assert_next_eq!(sub, VectorDiff::Set { index: 2, value: 'c' });
+
+    /* // Another value, that is moved to the left.
+    ob.set(0, 'a');
+    assert_next_eq!(sub, VectorDiff::Remove { index: 1 });
+    assert_next_eq!(sub, VectorDiff::Insert { index: 0, value: 'a' });
+
+    // Another value, that is moved to the right.
+    ob.set(0, 'f');
+    assert_next_eq!(sub, VectorDiff::Remove { index: 0 });
+    assert_next_eq!(sub, VectorDiff::Insert { index: 2, value: 'f' });
+
+    // Another value, that is moved to the right-most position.
+    ob.set(0, 'h');
+    assert_next_eq!(sub, VectorDiff::Remove { index: 2 });
+    assert_next_eq!(sub, VectorDiff::Insert { index: 3, value: 'h' });
+
+    // Same operation, at another index, just for fun.
+    ob.set(2, 'f');
+    assert_next_eq!(sub, VectorDiff::Remove { index: 0 });
+    assert_next_eq!(sub, VectorDiff::Insert { index: 1, value: 'f' });
+
+    // Items in the vector have been updated and are not sorted.
+    assert_eq!(*ob, vector!['h', 'e', 'f', 'g']); */
```

Comment:
Why is it commented?

Reply:
Looks like I didn't finish the tests after finding the issue you fixed in #49. Updated.